### ISSUE TITLE
IonIntlTelInputValidators

### ISSUE THREAD

**Issue by psenechal** [azzamasghar1/ion-intl-tel-input#37]
I noticed that using IonIntlTelInputValidators.phone as a validator also requires that a value be present (i.e. not null) which negates the option of using the validator on an optional field.

Is there a way to use the validator to check the validity of the number, but allow the input field to be null in case it's an optional field?

**Comment by azzamasghar1** [comment]
You are correct. I guess we could remove the error from field if its Null and if someone wants it to be required than they can add a required validator along with the phone validator of course.

**Comment by psenechal** [comment]
In my opinion, that would be the ideal solution so we can still get phone number validation AND required validation if needed, but also allow for an optional value. Thanks for considering the change.

**Comment by sjdrew** [comment]
I would agree, same situation for me.

**Comment by psenechal** [comment]
Hey Azzam,
Would this be a terribly difficult change to put in and release a new version? If so, I can attempt to create a pull request that you would just have to merge, but I'm not horribly familiar with google-libphonenumber.
Thanks!

**Comment by psenechal** [comment]
I was able to get around the issue by adding the following, but it would still be good to have this option:

```
this.form.get('phoneNumber')?.valueChanges
      .pipe(
        untilDestroyed(this),
        distinctUntilChanged((a, b) => JSON.stringify(a) === JSON.stringify(b))
      )
      .subscribe(value => {
        if (value) {
          this.form.get('phoneNumber')?.setValidators([IonIntlTelInputValidators.phone]);
        } else {
          this.form.get('phoneNumber')?.clearValidators();
        }

        this.form.get('phoneNumber')?.updateValueAndValidity();
      });
```
